src/HealthGPS.Core/exception.cpp

```diff
@@ -0,0 +1,28 @@
+#include "exception.h"
+
+#include <fmt/format.h>
+#include <iostream>
+
+namespace hgps::core {
+
+HgpsException::HgpsException(const std::string &what_arg, const source_location location)
+    : std::runtime_error{what_arg}, location_{location} {
+    what_arg_ = fmt::format("{}:{}: {}", file_name(), line(), std::runtime_error::what());
```

Comment:
I've only just noticed this, but how come you're passing `what_arg` to the parent constructor and then constructing a separate formatted string to give the extra source location info? Why doesn't the formatted string include `what_arg`? Surely we want this message to be visible to users. Why don't you construct a formatted string containing `what_arg` and the source location info, pass it to `std::runtime_error`'s constructor and drop the custom `what()` method?